src/client.rs

```diff
@@ -145,55 +149,62 @@ impl Client {
         Ok(txt)
     }
 
-    /// Calls the embedding endpoint.
-    ///
-    /// ## Arguments:
-    ///
-    /// * `req` - An instance of [`embedding::Request`]
-    ///
-    /// Returns a [`embedding::Response`]. A 200 (Ok) status code is expected from the Prediction Guard api. Any other status code
-    /// is considered an error.
-    pub async fn embedding(&self, req: &embedding::Request) -> Result<embedding::Response> {
-        let url = format!("{}{}", &self.inner.server, embedding::PATH);
+    pub async fn retrieve_model_list(&self, capability: String) -> Result<Vec<String>> {
```

Comment:
Public method should probably have some comments for documentation?

Reply:
Added in the new commit.